storm-core/src/jvm/org/apache/storm/topology/base/BaseWindowedBolt.java

```diff
@@ -255,6 +255,22 @@ public BaseWindowedBolt withTimestampField(String fieldName) {
     }
 
     /**
+     * Specify a stream id on which late tuples are going to be emitted. When using this method,
+     * {@link Config#TOPOLOGY_BOLTS_TUPLE_TIMESTAMP_FIELD_NAME} should be defined on a per component basis before
+     * calling this method, otherwise {@link IllegalArgumentException} will be thrown.
+     *
+     * @param streamId the name of the stream used to emit late tuples on
+     */
+    public BaseWindowedBolt withLateTupleStream(String streamId) {
+        if (!windowConfiguration.containsKey(Config.TOPOLOGY_BOLTS_TUPLE_TIMESTAMP_FIELD_NAME)) {
```

Comment:
On re-thinking, can this validation be moved to `initWindowManager` or `validate` so that we don't impose an ordering in the builder?